sdk/identity/identity/Troubleshooting.md

```diff
@@ -0,0 +1,241 @@
+## Troubleshooting Azure Identity Authentication Issues
+
+The Azure Identity SDK offers various `TokenCredential` implementations. These implementations tend to throw `CredentialUnavailableError` and `AuthenticationRequired` exceptions.
+The `CredentialUnavailableError` indicates that the credential cannot execute in the current environment setup due to lack of required configuration.
+The `AuthenticationRequiredError` indicates that the credential was able to run/execute but ran into an authentication issue from the server's end. This can happen due to invalid configuration/details passed in to the credential at construction time.
+This troubleshooting guide covers mitigation steps to resolve these exceptions thrown by various `TokenCredential` implementations in the Azure Identity Typescript client library.
```

Comment:
This introductory paragraph feels a bit loaded. I’m wondering, can we make this just an introductory paragraph without expanding on the errors, then have a first section in which we can talk about each error? What do you think?

Reply:
Aligning like .NET's